### ISSUE TITLE
[Feature Request] GPG Signing Support

### ISSUE THREAD

**Issue by OdinVex** [Murmele/Gittyup#121]
It'd be extremely helpful to have GPG-signing support, to verify uploads.

**Comment by dima-altinity** [comment]
+1 from me

**Comment by MeerMusik** [comment]
Yes please. I am currently looking for an Alternative to SmartGit. Unfortunately, Gittyup is unusable for me without this Feature, as I need to sign all my commits. Gittyup ignores all the aliases like `cm -s -S` `commit -s -S` set in the gitconfig File.

Thank you in advance!

**Comment by OdinVex** [comment]
> Yes please. I am currently looking for an Alternative to SmartGit. Unfortunately, Gittyup is unusable for me without this Feature, as I need to sign all my commits. Gittyup ignores all the aliases like `cm -s -S` `commit -s -S` set in the gitconfig File.
> 
> Thank you in advance!

The Flatpak version won't use your git config. You need to find 'your' git config in the Flatpak user files and edit it to autosign, or if you manually install it as per your distro you can edit your git config to autosign, and it will. Or you can edit the src to add signing. I require autosigning myself.

**Comment by MeerMusik** [comment]
Thanks for the Tip but I am running Gittyup on Windows 10 22H2. Not a Flatpak, Docker or whatever variant.

**Comment by OdinVex** [comment]
> Thanks for the Tip but I am running Gittyup on Windows 10 22H2. Not a Flatpak, Docker or whatever variant.

Find the actual git executable used by Gittyup and if it uses a 'local' git, then modify the config to add auto-signing. When I find time, I'll look at adding it in this week, you can then pull src and compile if needed before the next version release.

**Comment by OdinVex** [comment]
I've added support for automatic signing off **and** gpg signing as well. Simple checkboxes in the Options dialog. I'll push a PR after I get some translations for each language in.

**Comment by OdinVex** [comment]
Edit: I forgot that last year I modified Gittyup to remove code that worked with commits to use my installation's git instead, to specifically support signoff and GPG because libgit2 does support git_commit_create_with_signature, but that is required to be implemented via the software. I implemented it using host git. It was a hackish work-around I forgot about. It works, but I don't think Gittyup wants to use local git installations...even though that would bring 'your installed version of git' support. Sorry I got everyone's hopes up on that. My solution isn't a publicly-shippable one.

**Comment by WrichikBasu** [comment]
@OdinVex Maybe you could throw some light on what changes you made to the source files that makes Gittyup use my installed version of Git, and thereby sign commits? Then interested users can actually build from source if they like.
